.github/workflows/ci.yml

```
@@ -0,0 +1,213 @@
+          - elixir: "1.14.5"
+            otp: "25.3.2"
+      fail-fast: false
+
+    permissions:
+      contents: read
+      packages: read
+
+    env:
+      MIX_ENV: "test"
+      FWUP_VERSION: "1.13.2"
+
+    steps:
+      - name: Install fwup
+        run: |
+          wget https://github.com/fwup-home/fwup/releases/download/v${FWUP_VERSION}/fwup_${FWUP_VERSION}_amd64.deb
+          sudo dpkg -i fwup_${FWUP_VERSION}_amd64.deb && rm fwup_${FWUP_VERSION}_amd64.deb
+
+      - name: Install libmnl
+        run: sudo apt-get update && sudo apt-get install -y --no-install-recommends libmnl-dev cmake
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Elixir
+        id: beam
+        uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{matrix.otp}}
+          elixir-version: ${{matrix.elixir}}
+
+      - name: Cache deps
+        id: cache-deps
+        uses: actions/cache@v4
+        env:
+          cache-name: cache-elixir-deps
+        with:
+          path: deps
+          key: ${{ runner.os }}-${{ steps.beam.outputs.elixir-version }}-${{ steps.beam.outputs.otp-version }}-mix-${{ env.cache-name }}-${{ hashFiles('**/mix.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ steps.beam.outputs.elixir-version }}-${{ steps.beam.outputs.otp-version }}-mix-${{ env.cache-name }}-
+
+      - name: Cache compiled build
+        id: cache-build
+        uses: actions/cache@v4
+        env:
+          cache-name: cache-compiled-build
+        with:
+          path: _build
+          key: ${{ runner.os }}-${{ steps.beam.outputs.elixir-version }}-${{ steps.beam.outputs.otp-version }}-mix-${{ env.cache-name }}-${{ hashFiles('**/mix.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ steps.beam.outputs.elixir-version }}-${{ steps.beam.outputs.otp-version }}-mix-${{ env.cache-name }}-
+            ${{ runner.os }}-${{ steps.beam.outputs.elixir-version }}-${{ steps.beam.outputs.otp-version }}-mix-
+
+      - name: Clean to rule out incremental build as a source of flakiness
+        if: github.run_attempt != '1'
+        run: |
+          mix deps.clean --all
+          mix clean
+        shell: sh
+
+      - name: Install dependencies
+        run: mix deps.get
+
+      - name: Compiles without warnings
+        run: mix compile --warnings-as-errors
+
+      - name: Run tests
+        run: mix test
+
+  dialyzer-credo-and-more:
+    runs-on: ubuntu-latest
+
+    name: Dialyzer, Credo, and More
+
+    permissions:
+      contents: read
+      packages: read
+
+    env:
+      MIX_ENV: "test"
+
+    steps:
+      - name: Install libmnl
+        run: sudo apt-get update && sudo apt-get install -y --no-install-recommends libmnl-dev cmake
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Elixir
+        id: beam
+        uses: erlef/setup-beam@v1
+        with:
+          otp-version: 27.3.4
+          elixir-version: 1.18.4
```

```suggestion
          version-file: .tool-versions
          version-type: strict
```
I think we should rely on the version specified here rather than hard code

---

Ha, funny, I didn't even think of that because I forgot we have it.

---

Wait, we don't have one! Guess I'll need to add it.